x/minfee/upgrade_test.go

```diff
@@ -0,0 +1,128 @@
+package minfee_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/x/minfee"
+
+	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
+	v2 "github.com/celestiaorg/celestia-app/pkg/appconsts/v2"
+	"github.com/celestiaorg/celestia-app/test/util"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	version "github.com/tendermint/tendermint/proto/tendermint/version"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func TestUpgradeAppVersion(t *testing.T) {
+	testApp, _ := setupTestApp(t, 3)
+
+	supportedVersions := []uint64{v1.Version, v2.Version}
+
+	require.Equal(t, supportedVersions, testApp.SupportedVersions())
+
+	ctx := testApp.NewContext(true, tmproto.Header{
+		Version: version.Consensus{
+			App: 1,
+		},
+	})
+	testApp.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{
+		Height:  2,
+		Version: version.Consensus{App: 1},
+	}})
+
+	// app version should not have changed yet
+	require.EqualValues(t, 1, testApp.AppVersion())
+
+	// global min gas price should not have been set yet
+	gotBefore, err := testApp.ParamsKeeper.Params(ctx, &proposal.QueryParamsRequest{
+		Subspace: minfee.ModuleName,
+		Key:      string(minfee.KeyGlobalMinGasPrice),
+	})
```

Comment:
I think it's because the subspace exists? I tried querying it with multiple random keys and I got the same results back
<img width="323" alt="Screenshot 2024-03-18 at 20 04 43" src="https://github.com/celestiaorg/celestia-app/assets/84185790/303e157f-5a5f-4df0-8828-608cb439e74c">

Reply:
when subspace doesn't exist it returns nil and throws error